tachyon/crypto/hashes/sponge/poseidon/poseidon_sponge_base.h

```diff
@@ -136,7 +136,10 @@ struct PoseidonSpongeBase : public FieldBasedCryptographicSponge<Derived> {
   bool Absorb(const T& input) {
     std::vector<F> elements;
     if (!SerializeToFieldElements(input, &elements)) return false;
+    return Absorb(absl::MakeConstSpan(elements));
+  }
 
+  bool Absorb(absl::Span<const F> input) {
```

Comment:
I think this is not a good approach since the user has to know the fact that we have to explicitly call MakeConstSpan to avoid Serializing to Field elements. Can't we do this better with SFINAE?